csrc/preseg_passes/move_pad.cpp

```diff
@@ -398,93 +399,93 @@ void propagatePads(Fusion* fusion) {
-    if (tv->isFusionOutput()) {
+    if (pad_inp->isFusionOutput()) {
       continue;
     }
 
     // check for pad_dependencies to verify that 'p' can be moved before 'def'.
     std::unordered_set<Val*> pad_inputs;
-    for (Val* val : p->inputs()) {
-      if (val == p->in() || val->isConst()) {
+    for (Val* pad_inp : pad->inputs()) {
+      if (pad_inp == pad->in() || pad_inp->isConst()) {
         continue;
       }
-      pad_inputs.insert(val);
+      pad_inputs.insert(pad_inp);
     }
     std::unordered_set<Val*> pad_dependencies =
         DependencyCheck::getAllDependentVals(pad_inputs);
 
-    Expr* def = p->in()->definition();
+    Expr* def_of_pad_in = pad->in()->definition();
     Val* new_out = nullptr;
 
-    if (auto* uop = dynamic_cast<UnaryOp*>(def)) {
+    if (auto* uop = dynamic_cast<UnaryOp*>(def_of_pad_in)) {
       // check if unary op type is compatible for zero pad propagation.
       if (!zeroIsFixedPoint(uop->getUnaryOpType())) {
         continue;
       }
-      std::vector<Val*> new_pad_inputs = maybeMovePadBeforeDefinition(
-          tv, pad_dependencies, std::ref(stack), std::ref(simple_pad_set));
+      std::vector<Val*> outputs_of_moved_pad = maybeMovePadBeforeDefinition(
+          pad_inp, pad_dependencies, std::ref(stack), std::ref(simple_pad_set));
       // stop when move pad fails.
-      if (new_pad_inputs.empty()) {
+      if (outputs_of_moved_pad.empty()) {
         continue;
       }
       // update new outputs.
-      new_out =
-          ops::newValLike(new_pad_inputs[0], uop->out()->getDataType().value());
+      new_out = ops::newValLike(
+          outputs_of_moved_pad[0], uop->out()->getDataType().value());
       IrBuilder::create<UnaryOp>(
-          uop->getUnaryOpType(), new_out, new_pad_inputs[0]);
-    } else if (auto* bop = dynamic_cast<BinaryOp*>(def)) {
+          uop->getUnaryOpType(), new_out, outputs_of_moved_pad[0]);
+    } else if (auto* bop = dynamic_cast<BinaryOp*>(def_of_pad_in)) {
       // check if unary op type is compatible for zero pad propagation.
       if (!zeroIsIdentity(bop->getBinaryOpType())) {
         continue;
       }
       // check for broadcast on padded axis.
-      if (hasBroadcastOnAny(p->getPaddedAxes(), bop->inputs())) {
+      if (hasBroadcastOnAny(pad->getPaddedAxes(), bop->inputs())) {
         continue;
       }
 
-      std::vector<Val*> new_pad_inputs = maybeMovePadBeforeDefinition(
-          tv, pad_dependencies, std::ref(stack), std::ref(simple_pad_set));
+      std::vector<Val*> outputs_of_moved_pad = maybeMovePadBeforeDefinition(
+          pad_inp, pad_dependencies, std::ref(stack), std::ref(simple_pad_set));
       // stop when move pad fails.
-      if (new_pad_inputs.empty()) {
+      if (outputs_of_moved_pad.empty()) {
         continue;
       }
 
-      new_out =
-          ops::newOutputTV(new_pad_inputs, bop->out()->getDataType().value());
+      new_out = ops::newValLike(pad->output(0), pad->output(0)->dtype());
```

Comment:
I think you mean this line is where we have the logic change?

we switched from using `outputs_of_moved_pad->output(0)` to `pad->output(0)`

Reply:
Yes